salt/modules/zypper.py

```diff
@@ -628,16 +628,16 @@ def install(name=None,
         log.info('Targeting repo {0!r}'.format(fromrepo))
     else:
         fromrepoopt = ''
+    cmd_install = ['zypper', '--no-refresh', '--non-interactive', 'install',
+                   '--name', '--auto-agree-with-licenses']
+    if downloadonly:
+        cmd_install.append('--download-only')
+    if fromrepo:
+        cmd_install.extend(fromrepoopt)
     # Split the targets into batches of 500 packages each, so that
     # the maximal length of the command line is not broken
     while targets:
-        cmd = ['zypper', '--non-interactive', 'install', '--name',
-               '--auto-agree-with-licenses']
-        if downloadonly:
-            cmd.append('--download-only')
-
-        if fromrepo:
-            cmd.extend(fromrepoopt)
+        cmd = cmd_install
```

Comment:
This is a bug! Such assign makes only a pointer, not an isolated copy of the variable! That means that each modification to a `cmd` will be also in the `cmd_install`!!

@thatch45, I propose to carefully review merges, because this one breaks everything. :(